cli/mender-artifact/partition.go

```diff
@@ -42,13 +42,20 @@ type partition struct {
 type sdimgFile []io.ReadWriteCloser
 
 func newSDImgFile(fpath string, modcands []partition) (sdimgFile, error) {
+	// NOTE: Closure that captures the partitons-to-be-deleted variable,
+	// and cleans ups all the unused partitions on exit.
+	var delPartition []int = nil // By default clean up everything.
```

Comment:
Is it removing all by default? It seems that it removes nothing if `delPartition` is empty.